### ISSUE TITLE
Fulcrum database corruption

### ISSUE THREAD

**Issue by DanielPX1** [cculianu/Fulcrum#115]
Hi, 

Since Fulcrum is becoming more popular among bitcoin community due to its great features, I would like to ask a question, once database is complete, is fulcrum still prone to database corruption? If it is, how can I ensure safe reboot of device without corrupting the database?

Are there any easy set of commands for raspberry pi users, who do not have much experience in Linux? Would be great to add for RaspiBolt guides etc. Also for me as a linux noob who managed to corrupt it in several ways... 

Thank you very much for response

**Comment by cculianu** [comment]
>  is fulcrum still prone to database corruption?

Yes, it is -- but I do intend to fix that.  It happens if it is busy processing a block and the process is killed during that  200 msec time window (which happens only once every 10 mins).  So it happens rarely -- if you kill the process then forcefully most of the time it won't happen since the probability is low it was processing a block JUST when you kill it.

>  how can I ensure safe reboot of device without corrupting the database?

Ensure the process receives a signal like SIGTERM, SIGHUP, SIGQUIT, etc.  *Not* SIGKILL.  You can do this by using `systemd` and setting it up as a `.service`.  Here is my example `fulc-btc-main.service` file:

```
[Unit]
Description=Fulcrum (BTC Mainnet)
After=bitcoind-btc-main.service
StartLimitBurst=2
StartLimitIntervalSec=20

[Service]
Type=exec
Restart=on-failure
RestartSec=5
User=fulcrumdaemon
ExecStart=@/home/fulcrumdaemon/btc/Fulcrum /home/fulcrumdaemon/fulcrum-btc-mainnet.conf --ts-format none
KillSignal=SIGINT
TimeoutStopSec=300
SyslogIdentifier=Fulc-btc-main

[Install]
WantedBy=multi-user.target
```

In the above example I created a user called `fulcrumdaemon` and put the fulcrum stuff in its directory.  Notice how the `KillSignal` is set to `SIGINT`!

---

> Are there any easy set of commands for raspberry pi users, who do not have much experience in Linux?

Hmm.  Yeah maybe people that write the guides can do this.. I have my hands full now fixing bugs but will leave the issue open.

>  Also for me as a linux noob who managed to corrupt it in several ways...

Ha ha yeah sorry to hear that.  :/

**Comment by DanielPX1** [comment]
Thank you for response, updated my service file and will try not to interrupt syncing process.

**Comment by cculianu** [comment]
In the interests of keeping issues down, this is, in a way, a duplicate of #41 . so I will close this issue.

However feel free to continue to ask questions here if you like since I monitor these threads.
